book/src/generalized-ownership.md

```diff
@@ -0,0 +1,284 @@
+# Generalized Ownership
+
+The current implementation of user-defined tokens cannot be exchanged by
+other user-defined programs.
+
+This design describes generalized ownership of countable items on the Solana
+blockchain. The goal of this design is to propose a single general-purpose
+program whose state can be used across user-defined programs to implement
+ownership and transfer.
+
+Many programs in Solana should be able to use a common instruction similar to
+`SystemInstruction::Transfer`.  Furthermore, the amounts, destinations and sources
+should be based on outputs of programs as well as specified in transactions,
+while maintaining the same security guarantees of transfer of native tokens.
+
+## Goal for this design
+
+The goal of this design is to make the following possible:
+
+In a single program, it should be possible to change and update ownership of
+arbitrary units of tokens implemented by other programs.
+
+Alice requests X amount of token implemented by program Token1, in
+exchange for Y amount of token implemented by program Token2.
+
+Bob offers X + 1 amount of token implemented by program Token1, in exchange for
+Y - 1 amount of token implemented by Token2.
+
+Dan pairs Bob and Alice, causing the resulting ownership changes:
+
+* Alice owns X of Token1,
+* Bob owns Y - 1 of Token2,
+* Dan owns 1 of Token1 and 1 of Token2,
+
+Another example:
+
+Alice offers X amount of Token1 to the winner of TicTacToe.
+
+Bob offers Y amount of Token2 to the winner of the same game of TicTacToe.
+
+The game charges a commission, the ceiling of a percentage of the tokens.  The
+winner takes ownership of both tokens minus the commission.  The commission is
+transferred to the address designated by the game.
+
+## Current System and Token programs
+
+The System program defines ownership of lamports to be identified by the public key
+that matches the account's pubkey. The`SystemInstruction::Transfer` instruction checks that
+the identity of the spending account has a valid signature before completing the
+transfer.
+
+The Token program defines ownership of the token to be identified by the public key
+that matches the account's pubkey, and `Token::Transfer` instruction checks
+that the identity of the spending account has a valid signature before
+completing the transfer.
+
+Both of these programs implement almost the same mechanism.
+ 
+## General idea
+
+For items to have externally countable ownership, they must rely on the external
```

Comment:
This looks like the first use of "externally countable ownership". What does it mean?